### PR TITLE
fix: errors in mock oauth2 app export from test folder

### PR DESCRIPTION
**Pull request by deepakrkris**
fixes: https://github.com/strongloop/loopback-next/issues/5380

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: extensions/authentication-passport/src/mock-oauth2-social-app.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import bodyParser from 'body-parser';
 import express from 'express';
```
Comment (raymondfeng) [inline extensions/authentication-passport/src/mock-oauth2-social-app.ts:21]: It might be better to have it under `src/fixtures/`.

**Comment by raymondfeng** [comment]
Maybe we should find a better way to share the mock application, such as its package. Exporting a mock app from the `@loopback/authentication-passport` module is anti-pattern for LoopBack packages.

**Comment by deepakrkris** [comment]
> authentication-passport

@raymondfeng `authentication-passport` is not a core package, so I think this is fine atleast for now. But there is a story  to migrate `loopback-component-oauth2` for which I can add some features and move this into it's own package.

**Comment by raymondfeng** [comment]
> @raymondfeng authentication-passport is not a core package, so I think this is fine atleast for now. But there is a story to migrate loopback-component-oauth2 for which I can add some features and move this into it's own package.

The extension packages follow the same convention as core ones for consistency.

@bajtos What do you think?

**Comment by bajtos** [comment]
I agree with Raymond that it's a bad practice for packages to export test fixtures. Ideally, we should be excluding the entire `dist/__tests__` directory from our npm packages.

What are you @deepakrkris trying to achieve here? Why is `authentication-passport` exporting a mock application?

If you are trying to share test setup between multiple packages, then please follow the approach we are using for our `repository`-related code:
- `packages/repository` is providing the runtime behavior
- `packages/repository-tests` is exporting a shared test suite. It also comes with a local test suite that's running the shared test suite against the `memory` connector
- `acceptance/repository-mysql` and other packages are running the shared test suite against different connectors

In your case, perhaps you should create a new (internal/private?) package providing the mock application, and then add it to dev-dependencies of your projects.

Thoughts?

**Comment by bajtos** [comment]
Having wrote the above, maybe it's ok to land the changes in this PR as a short-term fix and then follow up with a refactoring to remove the mock app from exports.

**Comment by raymondfeng** [comment]
> In your case, perhaps you should create a new (internal/private?) package providing the mock application, and then add it to dev-dependencies of your projects.

I like the idea. Let's add a `fixtures` top-level directory under `loopback-next` and configure `lerna.json` to add the new location. We can move the mock app to be under that and declare devDepedencies for its consumer packages.

**Comment by raymondfeng** [comment]
@deepakrkris Let me know if you need some help to set it up.

**Comment by deepakrkris** [comment]
I am going to implement comments from @bajtos and @raymondfeng , then submit this PR for review again.

**Comment by deepakrkris** [comment]
@bajtos @raymondfeng please review

File: extensions/authentication-passport/tsconfig.json [hunk 2]
```diff
@@ -7,9 +7,13 @@
     "composite": true
   },
   "include": [
-    "src"
+    "src",
+    "../../fixtures/mock-oauth2-provider/src/mock-oauth2-social-app.ts"
```
Comment (raymondfeng) [inline extensions/authentication-passport/tsconfig.json:11]: This is wrong. We should list `@loopback/mock-oauth2-provider` as devDependency.

File: fixtures/mock-oauth2-provider/LICENSE [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Copyright (c) IBM Corp. 2017,2019.
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/LICENSE:1]: Please fix the header - I believe the year should be `2020`.

File: fixtures/mock-oauth2-provider/package.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@loopback/mock-oauth2-provider",
+  "version": "0.0.1",
+  "description": "",
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:4]: Add some description?

File: fixtures/mock-oauth2-provider/package.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/strongloop/loopback-next.git",
+    "directory": "fixtures/mock-oauth2-provider"
+  },
+  "dependencies": {
+    "passport": "^0.4.1",
+    "tslib": "^1.11.2",
+    "util-promisifyall": "^1.0.6"
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:36]: Is it used somewhere in the code?

File: fixtures/mock-oauth2-provider/package.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "devDependencies": {
+    "@loopback/build": "^5.3.1",
+    "@loopback/eslint-config": "^6.0.6",
+    "tslib": "^1.11.2",
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:41]: This is a duplicate of regular dep.

File: fixtures/mock-oauth2-provider/package.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    "@types/passport-oauth2": "^1.4.9",
+    "@types/qs": "^6.9.2",
+    "axios": "^0.19.2",
+    "body-parser": "^1.19.0",
+    "express": "^4.17.1",
+    "form-data": "^3.0.0",
+    "jsonwebtoken": "^8.5.1",
+    "lodash": "^4.17.15",
+    "passport-http": "^0.3.0",
+    "passport-oauth2": "^1.5.0"
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:56]: I believe some of the `devDependencies` should be moved into `dependencies` if they are used in the code of the mock app (not tests).

File: extensions/authentication-passport/tsconfig.json (hunk 2 above)
Comment (deepakrkris) [reply extensions/authentication-passport/tsconfig.json:11]: agreed

File: fixtures/mock-oauth2-provider/LICENSE (hunk 3 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/LICENSE:1]: agreed

File: fixtures/mock-oauth2-provider/package.json (hunk 7 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json:51]: agreed

File: fixtures/mock-oauth2-provider/package.json (hunk 6 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json]: agreed

File: fixtures/mock-oauth2-provider/package.json (hunk 5 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json]: agreed, will remove

File: fixtures/mock-oauth2-provider/package.json (hunk 4 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json]: agreed

File: fixtures/mock-oauth2-provider/package.json [hunk 8]
```diff
@@ -54,5 +49,11 @@
     "lodash": "^4.17.15",
     "passport-http": "^0.3.0",
     "passport-oauth2": "^1.5.0"
+  },
+  "devDependencies": {
+    "@loopback/build": "^5.3.1",
+    "@loopback/eslint-config": "^6.0.6",
+    "tslib": "^1.11.2",
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:56]: This needs to be in `dependencies`.

File: fixtures/mock-oauth2-provider/package.json [hunk 9]
```diff
@@ -33,12 +33,7 @@
   "dependencies": {
     "passport": "^0.4.1",
     "tslib": "^1.11.2",
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:35]: The latest version should be `2.0.0` now.

File: fixtures/mock-oauth2-provider/package.json (hunk 8 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json:56]: @raymondfeng all loopback packages refer only tslib 1.12.0 as of now

File: fixtures/mock-oauth2-provider/package.json (hunk 9 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json:35]: @raymondfeng all loopback packages refer only tslib 1.12.0 as of now

File: fixtures/mock-oauth2-provider/package.json (hunk 8 above)
Comment (deepakrkris) [reply fixtures/mock-oauth2-provider/package.json:56]: agreed

File: fixtures/mock-oauth2-provider/package.json [hunk 10]
```diff
@@ -31,23 +31,16 @@
     "directory": "fixtures/mock-oauth2-provider"
   },
   "dependencies": {
-    "passport": "^0.4.1",
-    "tslib": "^1.11.2",
+    "tslib": "^1.12.0",
```
Comment (raymondfeng) [inline fixtures/mock-oauth2-provider/package.json:34]: The version should be `2.0.0`.

File: fixtures/mock-oauth2-provider/package.json (hunk 9 above)
Comment (raymondfeng) [reply fixtures/mock-oauth2-provider/package.json]: No, you need to rebase to master. It's all `2.0.0` now.

**Comment by raymondfeng** [comment]
@deepakrkris Please squash the commits into one before landing it.

File: docs/site/MONOREPO.md [hunk 11]
```diff
@@ -55,6 +55,7 @@ The [loopback-next](https://github.com/strongloop/loopback-next) repository uses
...
 | [tsdocs](https://github.com/strongloop/loopback-next/tree/master/packages/tsdocs)                                             | @loopback/tsdocs                           | An internal package to generate api docs using Microsoft api-extractor and api-documenter                                                                                                                                                                                                        |
+| [fixtures/mock-oauth2-provider](https://github.com/strongloop/loopback-next/tree/master/fixtures/mock-oauth2-provider)        | _(private)_                                | An internal app to mock the OAuth2 authorization flow login with a social app like facebook, google etc                                                                                                                                                                                          |
```
Comment (bajtos) [inline docs/site/MONOREPO.md:58]: See https://github.com/strongloop/loopback-next/blob/1083430cee68d11b6328d32e285e02afe4550c1c/docs/site/MONOREPO.md#L6

```
<!-- PLEASE KEEP THE TABLE ROWS SORTED ALPHABETICALLY BY PACKAGE NAME-->
```

**Pull request merged**